patch/driver-arista-mmcblk-not-working-on-AMD-platforms.patch

```diff
@@ -0,0 +1,32 @@
+amd/mmc: mmcblk not working on some AMD platforms
+
+ADMA and ADMA-64 seem to be broken on AMD. This patch enables the
```

Comment:
“seem” sounds quite vague. What is the actual problem. What exact device was the problem seen with. What is the failure behavior?

Reply:
We diagnosed this issue mostly on products with an `AMD GX-424CC SOC` cpu using an emmc as the storage device.

The issue happens infrequently and only on devices that have often run for a while.
It is not a trivial problem to reproduce in the lab.

At some point the kernel starts complaining with lots of messages like the ones below.
```
mmc0: Timeout waiting for hardware interrupt.
sdhci: =========== REGISTER DUMP (mmc0)===========
sdhci: Sys addr: 0x00000078 | Version:  0x00001002
sdhci: Blk size: 0x00007200 | Blk cnt:  0x00000078
sdhci: Argument: 0x000ab148 | Trn mode: 0x0000003b
sdhci: Present:  0x01ff0001 | Host ctl: 0x00000019
sdhci: Power:    0x0000000f | Blk gap:  0x00000000
sdhci: Wake-up:  0x00000000 | Clock:    0x0000fa07
sdhci: Timeout:  0x0000000c | Int stat: 0x00000000
sdhci: Int enab: 0x02ff008b | Sig enab: 0x02ff008b
sdhci: AC12 err: 0x00000002 | Slot int: 0x000000ff
sdhci: Caps:     0x75fec8b2 | Caps_1:   0x00002501
sdhci: Cmd:      0x0000123a | Max curr: 0x00c80064
sdhci: Host ctl2: 0x00000000
sdhci: ADMA Err: 0x00000000 | ADMA Ptr: 0x000000020f97b20c
sdhci: ===========================================
mmcblk0: error -110 sending status command, retrying
mmcblk0: error -110 sending status command, retrying
mmcblk0: error -110 sending status command, aborting
mmc0: cache flush error -110
mmc0: tried to reset card, got error -110
blk_update_request: I/O error, dev mmcblk0, sector 700744
blk_update_request: I/O error, dev mmcblk0, sector 700752
...
```
At this point the storage is unreliable (comes and go) and obviously remounted in read-only.
There is no way to put the device back into a stable state and remounting it rw in SONiC.
Reboot is then the only option which is undesirable.

Arista's EOS ran into the same issue and our kernel team fixed it via a patch series.
The 2 patches in this PR are part of this series.

Reply:
Can you add this info (at minimum, the kernel logs) into the patch itself?

Reply:
Bump. Can this info be added into this patch?

Reply:
I added the info on in the patch